framework/web/Request.php

```diff
@@ -1097,8 +1187,8 @@ public function getContentType()
     public function getAcceptableLanguages()
     {
         if ($this->_languages === null) {
-            if (isset($_SERVER['HTTP_ACCEPT_LANGUAGE'])) {
-                $this->_languages = array_keys($this->parseAcceptHeader($_SERVER['HTTP_ACCEPT_LANGUAGE']));
+            if ($this->headers->get('Accept-Language')) {
```

Comment:
I think clear code is worth more. This path is only done once per request so the performance difference will be negligible.